### DISCUSSION TITLE
Eloquent encrypted relationship possible?

### DISCUSSION THREAD

**Discussion by flashmediasolutions** [laravel/framework#48049] [Q&A]
**Looking for any possible insight or help to understand what is possible within Laravel and using FilamentPHP for backend.**

Here is an example scenario:

There is a `phoneNumber` model and a `store` model.

A `store` `belongsTo` `phoneNumbers` and a `phoneNumber` `hasMany` `stores`

I understand the traditional creation of foreign id's and relationships.

My issue is that, in this example, the phone numbers are not sensitive, but store details and the relationship between the store and phone number is. 

Currently I encrypt the `phoneNumber` raw value in the `store` table. Then when I am pulling up the store, I decrypt the `phoneNumber` and search in the `phoneNumber` model to find the match.

So in the `store` model instead of using:
```
public function phoneNumber(): BelongsTo
{
   return $this->belongsTo(PhoneNumber::class);
}
```
I use:
```
public function getPhoneNumberAttribute()
{
   return PhoneNumber::where('phone_number', $this->phone_number_decrypted);
}
```

I want to obfuscate the relationship, between the store and phoneNumber, but maintain access to the eloquent query ability with the phone number and its relationships.

Is there any workaround for this?

Thanks for any insight!

**Comment by poing** [comment]
Yes, it's possible.  Here's a [how-to](https://poing.gitlab.io/sandstorm/blog/2024/12/09/encrypting-relationships/) explanation.
